planner/core/casetest/rule/testdata/join_reorder_suite_out.json

```diff
@@ -782,6 +782,167 @@
+      {
+        "SQL": "select * from t1 left join t2 on t1.a=t2.a",
+        "Plan": [
+          "IndexHashJoin 12487.50 root  left outer join, inner:IndexLookUp, outer key:test.t1.a, inner key:test.t2.a, equal cond:eq(test.t1.a, test.t2.a)",
+          "├─TableReader(Build) 10000.00 root  data:TableFullScan",
+          "│ └─TableFullScan 10000.00 cop[tikv] table:t1 keep order:false, stats:pseudo",
+          "└─IndexLookUp(Probe) 12487.50 root  ",
+          "  ├─Selection(Build) 12487.50 cop[tikv]  not(isnull(test.t2.a))",
+          "  │ └─IndexRangeScan 12500.00 cop[tikv] table:t2, index:a(a) range: decided by [eq(test.t2.a, test.t1.a)], keep order:false, stats:pseudo",
+          "  └─TableRowIDScan(Probe) 12487.50 cop[tikv] table:t2 keep order:false, stats:pseudo"
+        ],
+        "Warning": null
+      },
+      {
+        "SQL": "select * from t1 right join t2 on t1.b=t2.b",
+        "Plan": [
+          "MergeJoin 12487.50 root  right outer join, left key:test.t1.b, right key:test.t2.b",
+          "├─Sort(Build) 9990.00 root  test.t1.b",
+          "│ └─TableReader 9990.00 root  data:Selection",
+          "│   └─Selection 9990.00 cop[tikv]  not(isnull(test.t1.b))",
+          "│     └─TableFullScan 10000.00 cop[tikv] table:t1 keep order:false, stats:pseudo",
+          "└─Sort(Probe) 10000.00 root  test.t2.b",
+          "  └─TableReader 10000.00 root  data:TableFullScan",
+          "    └─TableFullScan 10000.00 cop[tikv] table:t2 keep order:false, stats:pseudo"
+        ],
+        "Warning": null
+      },
+      {
+        "SQL": "select * from t1 right join t2 on t1.a=t2.a",
+        "Plan": [
+          "IndexHashJoin 12487.50 root  right outer join, inner:IndexLookUp, outer key:test.t2.a, inner key:test.t1.a, equal cond:eq(test.t2.a, test.t1.a)",
+          "├─TableReader(Build) 10000.00 root  data:TableFullScan",
+          "│ └─TableFullScan 10000.00 cop[tikv] table:t2 keep order:false, stats:pseudo",
+          "└─IndexLookUp(Probe) 12487.50 root  ",
+          "  ├─Selection(Build) 12487.50 cop[tikv]  not(isnull(test.t1.a))",
+          "  │ └─IndexRangeScan 12500.00 cop[tikv] table:t1, index:a(a) range: decided by [eq(test.t1.a, test.t2.a)], keep order:false, stats:pseudo",
+          "  └─TableRowIDScan(Probe) 12487.50 cop[tikv] table:t1 keep order:false, stats:pseudo"
+        ],
+        "Warning": null
+      },
+      {
+        "SQL": "select /*+ hash_join(t1) */ * from t1, t2",
+        "Plan": [
+          "HashJoin 100000000.00 root  CARTESIAN inner join",
+          "├─TableReader(Build) 10000.00 root  data:TableFullScan",
+          "│ └─TableFullScan 10000.00 cop[tikv] table:t2 keep order:false, stats:pseudo",
+          "└─TableReader(Probe) 10000.00 root  data:TableFullScan",
+          "  └─TableFullScan 10000.00 cop[tikv] table:t1 keep order:false, stats:pseudo"
+        ],
+        "Warning": null
+      },
+      {
+        "SQL": "select /*+ leading(t4, t3, t2, t1) */ * from t1, t2, t3, t4",
```

Comment:
Why hashjoin can't be disabled in this case?

Reply:
I'll take a look at this.

Reply:
`no` is not a valid value for a bool variable in TiDB, only `on`, `off`, `1`, `0`.

Reply:
@qw4990  My original meaning was that even in your PR, the hash join is not disabled when applying a `leading` hint with a `no_hash_join_hint`.  

see my comment on https://github.com/pingcap/tidb/pull/45538

Reply:
In my case, the `no_hash_join` only works for `t2` and `t3`. The top Hash Join is applied on `t1` and `result of (t2, t3, t4)`, so `no_hash_join(t2)` cannot take effect on it.
![image](https://github.com/pingcap/tidb/assets/7499936/8a100aa1-66e7-4c0e-a0df-a9035bd697eb)